src/features/allDataState/index.js

```diff
@@ -0,0 +1,297 @@
+  selectGroceriesBasketData,
+  selectProductsBasketData,
+  selectDishesBasketData,
+  selectSavedListBasketData,
+
+  (
+    selectGroceriesBasketData,
+    selectProductsBasketData,
+    selectDishesBasketData,
+    selectSavedListBasketData,
+  ) => {
+    const empty =
+      selectGroceriesBasketData.length === 0 &&
+      selectProductsBasketData.length === 0 &&
+      selectDishesBasketData.length === 0 &&
+      selectSavedListBasketData.length === 0
+
+    return empty
+  },
+)
+
+export const itemsDataStatus = createSelector(
+  selectGroceriesStatus,
+  selectProductsStatus,
+  selectDishesStatus,
+  selectSavedListStatus,
+  selectParentCategoriesStatus,
+  isBasketEmpty,
+
+  (
+    selectGroceriesStatus,
+    selectProductsStatus,
+    selectDishesStatus,
+    selectSavedListStatus,
+    selectParentCategoriesStatus,
+    isBasketEmpty,
+  ) => {
+    const dataStatus = [
+      selectGroceriesStatus,
+      selectProductsStatus,
+      selectDishesStatus,
+      selectSavedListStatus,
+      selectParentCategoriesStatus,
+    ]
+
+    let result = {
+      isError: false,
+      isLoading: true,
+      isLoaded: false,
+      isBasketEmpty: false,
+    }
+
+    let workingIsLoaded = true
+    result.isBasketEmpty = isBasketEmpty
+
+    dataStatus.forEach((singleStatus) => {
+      if (singleStatus === 'failed') {
+        result.isError = true
+      }
+
+      if (singleStatus === 'inProgress') {
+        result.isLoading = true
+      }
+
+      if (singleStatus !== 'succeeded') {
+        workingIsLoaded = false
+      }
+    })
+
+    result.isLoaded = workingIsLoaded
+
+    if (result.isLoaded) {
+      result.isLoading = false
+    }
+
+    return result
+  },
+)
+
+export const smartListStatus = createSelector(
+  selectGroceriesStatus,
+  selectProductsStatus,
+  selectDishesStatus,
+  selectSavedListStatus,
+  selectParentCategoriesStatus,
+  selectHistoryStatus,
+  selectSettingsStatus,
+
+  (
+    selectGroceriesStatus,
+    selectProductsStatus,
+    selectDishesStatus,
+    selectSavedListStatus,
+    selectParentCategoriesStatus,
+    selectHistoryStatus,
+    selectSettingsStatus,
+  ) => {
+    const dataStatus = [
+      selectGroceriesStatus,
+      selectProductsStatus,
+      selectDishesStatus,
+      selectSavedListStatus,
+      selectParentCategoriesStatus,
+      selectHistoryStatus,
+      selectSettingsStatus,
+    ]
+
+    let result = {
+      isError: false,
+      isLoading: true,
+      isLoaded: false,
+    }
+
+    let workingIsLoaded = true
+
+    dataStatus.forEach((singleStatus) => {
+      if (singleStatus === 'failed') {
+        result.isError = true
+      }
+
+      if (singleStatus === 'inProgress') {
+        result.isLoading = true
+      }
+
+      if (singleStatus !== 'succeeded') {
+        workingIsLoaded = false
+      }
+    })
+
+    result.isLoaded = workingIsLoaded
+
+    if (result.isLoaded) {
+      result.isLoading = false
+    }
+
+    return result
```

Comment:
zmieniłbym to trochę, tak żeby nie zwracać obiektu z isError, isLoading i isLoaded, a zwrócić stringa: 'error' / 'loading' / 'success'. Jeżeli jest error, to nie obchodzi Cie czy cos sie laduje. Jezeli sie zaladowalo poprawnie, to zakladasz, ze nie ma bledu, etc. Wtedy w komponencie po prostu robisz if status === 'error' to cośtam

Reply:
Rozumiem ale moim celem było aby nie robić sprawdzenia podczas weryfikacji  "if status === 'error' to cośtam" tylko if(data.isError) {
 cośtam
} 
widziałem ten schemat w react-query libka do api i zamierzam ją zastosować ;)
Pomijam.